Assets/Scripts/Enemy/EnemySpawner.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.AI;
+using UnityEngine.PlayerLoop;
+using Random = UnityEngine.Random;
+
+public class EnemySpawner : MonoBehaviour
+{
+    [SerializeField] private EnemyScriptableObjectList enemyList;
+    private Vector3 randomPoint;
+    private Coroutine spawnCoroutine = null;
+    [SerializeField] private int spawnCount;
+    [SerializeField] private float range;
+    private bool isRunning;
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        isRunning = false;
+    }
+    void Update()
+    {
```

Comment:
Wrap the things inside the Update method to a seperate method to increase code readability.